GsuiteAdmin update - 7 new commands (#33396)
demisto/content: Packs/GSuiteAdmin/Integrations/GSuiteAdmin/README.md

| GSuite.User.posixAccounts.shell | String | The path to the login shell for this account. | 
| GSuite.User.posixAccounts.gecos | String | The GECOS \(user information\) for this account. | 
| GSuite.User.posixAccounts.systemId | String | System identifier for which account Username or Uid apply to. | 
| GSuite.User.posixAccounts.primary | Boolean | If this is user's primary account within the SystemId. | 
| GSuite.User.posixAccounts.accountId | String | A POSIX account field identifier. | 
| GSuite.User.posixAccounts.operatingSystemType | String | The operating system type for this account. | 
| GSuite.User.sshPublicKeys.key | String | An SSH public key. | 
| GSuite.User.sshPublicKeys.expirationTimeUsec | String | An expiration time in microseconds since epoch. | 
| GSuite.User.sshPublicKeys.fingerprint | String | A SHA-256 fingerprint of the SSH public key. | 
| GSuite.User.aliases | Unknown | List of the user's alias email addresses. | 
| GSuite.User.nonEditableAliases | Unknown | List of the user's non-editable alias email addresses. These are typically outside the account's primary domain or sub-domain. | 
| GSuite.User.websites.value | String | The URL of the website. | 

> | GSuite.User.websites.primary | Boolean | If this is user's primary website or not. | 

Reviewer: ```suggestion
| GSuite.User.websites.primary | Boolean | If this is the user's primary website or not. | 
```

| GSuite.User.websites.type | String | The type or purpose of the website. For example, a website could be labeled as home or blog. Alternatively, an entry can have a custom type. | 
| GSuite.User.websites.customType | String | The custom type. Only used if the type is custom. | 
| GSuite.User.locations.type | String | The location type. | 
| GSuite.User.locations.customType | String | If the location type is custom, this property contains the custom value. | 
| GSuite.User.locations.area | String | Textual location. This is most useful for display purposes to concisely describe the location. For example, "Mountain View, CA", "Near Seattle". | 
| GSuite.User.locations.buildingId | String | Building identifier. | 
| GSuite.User.locations.floorName | String | Floor name/number. | 
| GSuite.User.locations.floorSection | String | Floor section. More specific location within the floor. For example, if a floor is divided into sections "A", "B", and "C", this field would identify one of those values. | 
| GSuite.User.locations.deskCode | String | Most specific textual code of individual desk location. | 

> | GSuite.User.keywords.type | String | Each entry can have a type which indicates standard type of that entry. For example, keyword could be of type occupation or outlook. In addition to the standard type, an entry can have a custom type and can give it any name. Such types should have the CUSTOM value as type and also have a customType value. | 

Reviewer: ```suggestion
| GSuite.User.keywords.type | String | Each entry can have a type which indicates the standard type of that entry. For example, keyword could be of type occupation or outlook. In addition to the standard type, an entry can have a custom type and can give it any name. Such types should have the CUSTOM value as type and also have a customType value. | 
```

| GSuite.User.keywords.customType | String | Custom Type. | 
| GSuite.User.keywords.value | String | Keyword. | 
| GSuite.User.isEnrolledIn2Sv | Boolean | Is enrolled in 2-step verification. | 
| GSuite.User.isEnforcedIn2Sv | Boolean | Is 2-step verification enforced. | 
| GSuite.User.includeInGlobalAddressList | Boolean | Indicates if the user's profile is visible in the G Suite global address list when the contact sharing feature is enabled for the domain. | 

> | GSuite.User.thumbnailPhotoUrl | String | Photo Url of the user. | 

Reviewer: ```suggestion
| GSuite.User.thumbnailPhotoUrl | String | Photo URL of the user. | 
```

            "suspended": "False",
            "archived": "False",
            "changePasswordAtNextLogin": "true",
            "ipWhitelisted": "False",
            "emails": [
                {
                "address": "nikolic@demistodev.com",
                "primary": "True"
                },
                {
                "address": "nikolic@demistodev.com.test-google-a.com"
                }
            ],
            "nonEditableAliases": ["nikolic@demistodev.com.test-google-a.com"],
            "customerId": "C02f0zfqw",
            "orgUnitPath": "/",
            "isMailboxSetup": "True",
            "includeInGlobalAddressList": "True",
            "thumbnailPhotoUrl": "//lh3.googleusercontent.com/a/ACg8ocJ85an5w7nMMFnC78mUqjuy2jhUHv2cSn7_-DeK0X93=mo",
            "thumbnailPhotoEtag": "BvGfkzKoKVD0NM7VdXdzkXDD-nHLkyMjheL_9Z5X0H0/iIfdCQy7n1Lau3YZCJL6xXPk3WE"
        }
    }
}
```

#### Human Readable Output

>### Updated User Details
>|Id|Customer Id|Primary Email|Change Password At Next Login|
>|---|---|---|---|
>| 104065793340555884068 | C02f0zfqw | nikolic@demistodev.com | true |

### gsuite-chromebrowserdevice-move-ou

***

> To move Chrome Browser devices assigned to an account from one organization unit to another, use the following POST request and include the access token obtained by following the "Authorization" section.

Reviewer: ```suggestion
To move Chrome browser devices assigned to an account from one organization unit to another, use the following POST request and include the access token obtained by following the "Authorization" section.
```

#### Base Command

`gsuite-chromebrowserdevice-move-ou`

##### Required Permissions
`https://www.googleapis.com/auth/admin.directory.device.chromebrowsers`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| customer_id | The customerId is a unique identifier of the customer's Google account. | Required | 

> | resource_ids | List of unique device IDs (comma separated) of Chrome Browser Devices to move. A maximum of 600 browsers may be moved per request. | Required | 

Reviewer: ```suggestion
| resource_ids |  A comma-separated list of unique device IDs of Chrome browser devices to move. A maximum of 600 browsers may be moved per request. | Required | 
```

| org_unit_path | Destination organization unit to move devices to. Full path of the organizational unit or its ID prefixed with 'id:'. | Required | 

#### Context Output

There is no context output for this command.

#### Command Example
```!gsuite-chromebrowserdevice-move-ou customer_id="test" resource_ids="11111" org_unit_path="/testing"```

#### Human Readable Output
> Chrome browser devices have been moved to the new organization unit ORGANIZATIONAL_UNIT_PATH.

### gsuite-chromebrowserdevice-list

***

> Retrieve all Chrome browser devices for an account or a specific Chrome browser device

Reviewer: ```suggestion
Retrieve all Chrome browser devices for an account or a specific Chrome browser device.
```

#### Base Command

`gsuite-chromebrowserdevice-list`

##### Required Permissions
`https://www.googleapis.com/auth/admin.directory.device.chromebrowsers`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |

> | customer_id | The customerId is a unique identifier of the customer's Google account. | Required | 

Reviewer: ```suggestion
| customer_id | The unique identifier of the customer's Google account. | Required | 
```

| device_id | The unique ID of the device. | Optional | 

> | order_by | Field according to which to order by. Possible values are: id, last_sync, machine_name, extension_count, policy_count, os_version, last_signed_in_user, annotated_user, annotated_location, annotated_asset_id, notes, browser_version_channel, org_unit, enrollment_date, save_browsing_clickthrough, platform_major_version, last_activity, browser_version_sortable, os_version_sortable. | Optional | 

Reviewer: ```suggestion
| order_by | Field by which to order. Possible values are: id, last_sync, machine_name, extension_count, policy_count, os_version, last_signed_in_user, annotated_user, annotated_location, annotated_asset_id, notes, browser_version_channel, org_unit, enrollment_date, save_browsing_clickthrough, platform_major_version, last_activity, browser_version_sortable, os_version_sortable. | Optional | 
```

| org_unit_path | The full path of the organizational unit or its unique ID. | Optional | 
| projection | Restrict information returned to a set of selected fields. "BASIC" - Includes only the basic metadata fields (i.e., the ones found in the columns of the admin console browser list as enumerated above). "FULL" - Includes all metadata fields (as enumerated above). Possible values are: BASIC, FULL. | Optional | 

> | query | Search string using the list page query language described in the section below (Filter Query Language). | Optional | 

Reviewer: There is no section called Filter Query Language in this doc. Perhaps you need to add a link here to an external doc?

> | sort_order | Whether to return results in ascending or descending order. Must be used with the orderBy parameter. "ASCENDING" - Ascending order. "DESCENDING" - Descending order. Possible values are: ASCENDING, DESCENDING. | Optional | 

Reviewer: ```suggestion
| sort_order | Whether to return results in ascending or descending order. Must be used with the orderBy parameter. Possible values are: ASCENDING, DESCENDING. | Optional | 
```

> | page_size | The maximum number of policy schemas to return, default and maximum are 100. Default is 100. | Optional | 

Reviewer: ```suggestion
| page_size | The maximum number of policy schemas to return. Default and maximum are 100. | Optional | 
```

> | page_token | Page Token. | Optional | 

Reviewer: ```suggestion
| page_token | Page token. | Optional | 
```

> | limit | Limit. | Optional | 

Reviewer: Limit of what? The number of devices per retrieval? The number of devices per page?

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |
| GSuite.ChromeBrowserDevices.deviceId | String | The unique Device ID for the device. | 

> | GSuite.ChromeBrowserDevices.kind | String | Device kind. | 

Reviewer: ```suggestion
| GSuite.ChromeBrowserDevices.kind | String | The type of device. | 
```

> | GSuite.ChromeBrowserDevices.osPlatform | String | Device OS Platform. | 
> | GSuite.ChromeBrowserDevices.osVersion | String | Device OS Version. | 
> | GSuite.ChromeBrowserDevices.machineName | String | Device Machine Name. | 
> | GSuite.ChromeBrowserDevices.lastRegistrationTime | String | Device Last Registration Time. | 
> | GSuite.ChromeBrowserDevices.lastActivityTime | String | Device Last Activity Time. | 
> | GSuite.ChromeBrowserDevices.osPlatformVersion | String | Device OS Platform Version. | 
> | GSuite.ChromeBrowserDevices.serialNumber | String | Device Serial Number. | 
> | GSuite.ChromeBrowserDevices.virtualDeviceId | String | Device Virtual ID. | 
> | GSuite.ChromeBrowserDevices.orgUnitPath | String | Device Organizational Unit Path. | 
> | GSuite.ChromeBrowserDevices.deviceIdentifiersHistory | Array | Device Identifier History. |

Reviewer: ```suggestion
| GSuite.ChromeBrowserDevices.osPlatform | String | Device operating system platform. | 
| GSuite.ChromeBrowserDevices.osVersion | String | Device operating system version. | 
| GSuite.ChromeBrowserDevices.machineName | String | Device machine name. | 
| GSuite.ChromeBrowserDevices.lastRegistrationTime | String | Device last registration time. | 
| GSuite.ChromeBrowserDevices.lastActivityTime | String | Device last activity time. | 
| GSuite.ChromeBrowserDevices.osPlatformVersion | String | Device operating system platform version. | 
| GSuite.ChromeBrowserDevices.serialNumber | String | Device serial number. | 
| GSuite.ChromeBrowserDevices.virtualDeviceId | String | Device virtual ID. | 
| GSuite.ChromeBrowserDevices.orgUnitPath | String | Device organizational unit path. | 
| GSuite.ChromeBrowserDevices.deviceIdentifiersHistory | Array | Device identifier history. |
```

> | GSuite.ChromeBrowserDevices.deviceIdentifiersHistory | Array | Device Identifier History. |

Reviewer: If this is an array, then is this a comma-separated list of .....

#### Command Example
```!gsuite-chromebrowserdevice-list customer_id=test```

#### Context Example
```json
{
    "GSuite": {
        "ChromeBrowserDevices": [
            {
                "deviceId": "test", 
                "kind": "admin#directory#browserdevice", 
                "osPlatform": "Windows", 
                "osVersion": "10.0.19045.4046", 
                "machineName": "WIN10-test", 
                "lastRegistrationTime": "2024-02-20T21:16:00.902Z", 
                "lastActivityTime": "2024-02-20T21:16:00.902Z", 
                "osPlatformVersion": "Windows 10", 
                "serialNumber": "test 00 00 00", 
                "virtualDeviceId": "11111-aca1116-1111-1111-6823faec8350", 
                "orgUnitPath": "/testing", 
                "deviceIdentifiersHistory": {}
            }
        ]
    }
}
```

#### Human Readable Output

>### Chrome Browser Device List
>|Device Id|Os Platform|Os Version|Machine Name|Serial Number|Org Unit Path|
>|---|---|---|---|---|---|
>| test | Windows | 10.0.19045.4046 | WIN10-test | test 00 00 00 | /testing |

### gsuite-policy-modify

***

> Delete multiple policy values that are applied to a specific group. All targets must have the same target format. That is to say that they must point to the same target resource and must have the same keys specified in additionalTargetKeyNames, though the values for those keys may be different. On failure the request will return the error details as part of the google.rpc.Status.

Reviewer: ```suggestion
Modifies multiple policy values that are applied to a specific group. All targets must have the same target format. That is to say that they must point to the same target resource and must have the same keys specified in additionalTargetKeyNames, though the values for those keys may be different. On failure the request will return the error details as part of the google.rpc.Status.
```

##### Required Permissions
`https://www.googleapis.com/auth/chrome.management.policy`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| customer_id | ID of the Google Workspace account or literal 'my_customer' for the customer associated with the request. | Required | 
| target_type | In case of Group use “POST…groups:batchModify”. In case of OrgUnit use “POST…orgunits:batchModify”. Possible values are: Group, OrgUnit. | Required | 
| policy_raw_json | String as JSON representation of the ModifyGroupPolicyRequest<br/>{<br/>    "requests": [<br/>        {<br/>            "policyTargetKey": {<br/>                "targetResource": "orgunits/03ph8a2z1kjba6k",<br/>                "additionalTargetKeys": {<br/>                    "app_id": "chrome:jldhpllghnbhlbpcmnajkpdmadaolakh"<br/>                }<br/>            },<br/>            "policyValue": {<br/>                "policySchema": "chrome.users.apps.InstallType",<br/>                "value": {<br/>                    "appInstallType": "BLOCKED"<br/>                }<br/>            },<br/>            "updateMask": "appInstallType"<br/>        }<br/>   ]<br/>}. | Optional | 

> | policy_field_json_entry_id | Entry Id of the file that contains JSON representation of the ModifyGroupPolicyRequest. | Optional | 

Reviewer: ```suggestion
| policy_field_json_entry_id | Entry ID of the file that contains JSON representation of the ModifyGroupPolicyRequest. | Optional | 
```

#### Context Output

There is no context output for this command.

#### Command Example
```!gsuite-policy-modify customer_id=test target_type=Group target_resource=1111111111 policy_schema_filter=chrome.users.apps.InstallType additional_target_keys="\"app_id\":\"chrome:1111111aaaaaaaaa\"" policy_schema=chrome.users.apps.InstallType policy_value=BLOCKED update_mask=appInstallType```

#### Human Readable Output
> Policy has been modified for the customer CUSTOMER_ID.

### gsuite-policy-schemas-list

***
Gets a list of policy schemas that match a specified filter value for a given customer.

#### Base Command

`gsuite-policy-schemas-list`

##### Required Permissions
`https://www.googleapis.com/auth/chrome.management.policy`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| customer_id | ID of the Google Workspace account or literal "my_customer" for the customer associated with the request. | Required | 

> | schema_name | Required when asking for a specific policy. The policy schema resource name to query. | Optional | 

Reviewer: ```suggestion
| schema_name | The policy schema resource name to query. Required when asking for a specific policy.  | Optional | 
```

| filter | The schema filter is used to find a particular schema based on fields like its resource name, description and additionalTargetKeyNames. | Optional | 

> | page_size | The maximum number of policy schemas to return, defaults to 100 and has a maximum of 1000. Default is 100. | Optional | 

Reviewer: ```suggestion
| page_size | The maximum number of policy schemas to return. Defaults to 100 and has a maximum of 1000. | Optional | 
```

> | page_token | Page Token. | Optional | 

Reviewer: ```suggestion
| page_token | Page token. | Optional | 
```

> | limit | Limit. | Optional | 

Reviewer: Limit of what? The number of devices per retrieval? The number of devices per page?

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |

> | GSuite.PolicySchema.name | String | Policy Name | 
> | GSuite.PolicySchema.policyDescription | String | Policy Description | 
> | GSuite.PolicySchema.definition | Array | Policy Definition. | 
> | GSuite.PolicySchema.fieldDescriptions | Array | Policy File Description. | 
> | GSuite.PolicySchema.notices | Array | Policy Notice. | 
> | GSuite.PolicySchema.supportUri | String | Policy Support URI. | 
> | GSuite.PolicySchema.schemaName | String | Policy Schema Name. | 
> | GSuite.PolicySchema.validTargetResources | Array | Policy ValidTarget Resources. | 
> | GSuite.PolicySchema.policyApiLifecycle | String | Policy API Lifecycle. | 
> | GSuite.PolicySchema.categoryTitle | String | Policy Category Title. | 
> | GSuite.PolicySchema.supportedPlatforms | String | Policy Supported Platforms. | 

Reviewer: ```suggestion
| GSuite.PolicySchema.name | String | Policy name. | 
| GSuite.PolicySchema.policyDescription | String | Policy description. | 
| GSuite.PolicySchema.definition | Array | Policy definition. | 
| GSuite.PolicySchema.fieldDescriptions | Array | Policy file description. | 
| GSuite.PolicySchema.notices | Array | Policy notice. | 
| GSuite.PolicySchema.supportUri | String | Policy support URI. | 
| GSuite.PolicySchema.schemaName | String | Policy schema name. | 
| GSuite.PolicySchema.validTargetResources | Array | Policy valid target resources. | 
| GSuite.PolicySchema.policyApiLifecycle | String | Policy API lifecycle. | 
| GSuite.PolicySchema.categoryTitle | String | Policy category title. | 
| GSuite.PolicySchema.supportedPlatforms | String | Policy supported platforms. | 
```

            ],
            "policyApiLifecycle": {
                "policyApiLifecycleStage": "API_CURRENT"
            },
            "categoryTitle": "Additional app settings",
            "supportedPlatforms": [
                "CHROME_OS"
            ]
        }
    ]
    }
}
```

#### Human Readable Output

>### Policy Schemas List
>|Name|Policy Description|Schema Name|
>|---|---|---|
>| customers/test/policySchemas/chrome.devices.test.appsconfig.AutoLaunchApp | Allows setting of the auto-launch app. | chrome.devices.test.appsconfig.AutoLaunchApp |
>| customers/test/policySchemas/chrome.users.appsconfig.AndroidAppsEnabled | Android apps on Chrome devices. | chrome.devices.test.appsconfig.AutoLaunchApp |

### gsuite-policy-resolve

***
Gets the resolved policy values for a list of policies that match a search query.

#### Base Command

`gsuite-policy-resolve`

##### Required Permissions
`https://www.googleapis.com/auth/chrome.management.policy`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| customer_id | ID of the Google Workspace account or literal "my_customer" for the customer associated with the request. | Required | 

> | policy_schema_filter | Specify a schema name to view a particular schema, for example: chrome.users.ShowLogoutButton, Please read https://developers.google.com/chrome/policy/guides/policy-schemas for details on schema namespaces. | Required | 

Reviewer: ```suggestion
| policy_schema_filter | Specify a schema name to view a particular schema, for example: chrome.users.ShowLogoutButton. See https://developers.google.com/chrome/policy/guides/policy-schemas for details on schema namespaces. | Required | 
```

> | target_type | Type of target, can be Group or Organizational Unit. Possible values are: Group, OrgUnit. | Required | 

Reviewer: ```suggestion
| target_type | Type of target. Possible values are: Group, OrgUnit. | Required | 
```

| target_resource | The target resource on which this policy is applied. The following resources are supported: Organizational Unit ("orgunits/{orgunit_id}") or Group ("groups/{groupId}"). | Required | 
| additional_target_keys | Map (key: string, value: string) containing the additional target key name and value pairs used to further identify the target of the policy. | Optional | 

> | page_size | The maximum number of policy schemas to return, defaults to 100 and has a maximum of 1000. Default is 100. | Optional | 

Reviewer: ```suggestion
| page_size | The maximum number of policy schemas to return. Defaults to 100 and has a maximum of 1000. | Optional | 
```

> | page_token | Page Token. | Optional | 

Reviewer: ```suggestion
| page_token | Page token. | Optional | 
```

> | limit | Limit. | Optional | 

Reviewer: Limit of what? The number of devices per retrieval? The number of devices per page?

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |

> | GSuite.Policy.targetKey | unknown | Policy Target Key | 
> | GSuite.Policy.value | unknown | Policy Value | 
> | GSuite.Policy.sourceKey | unknown | Policy Source Key | 
> | GSuite.Policy.addedSourceKey | unknown | Policy Added Source Key | 

Reviewer: ```suggestion
| GSuite.Policy.targetKey | unknown | Policy target key. | 
| GSuite.Policy.value | unknown | Policy value. | 
| GSuite.Policy.sourceKey | unknown | Policy source key. | 
| GSuite.Policy.addedSourceKey | unknown | Policy added source key. | 
```
